test/extensions/transport_sockets/tls/io_handle_bio_test.cc

```diff
@@ -24,6 +25,15 @@ class IoHandleBioTest : public testing::Test {
   NiceMock<Network::MockIoHandle> io_handle_;
 };
 
+TEST_F(IoHandleBioTest, WriteError) {
+  EXPECT_CALL(io_handle_, writev(_, 1))
+      .WillOnce(Return(testing::ByMove(
+          Api::IoCallUint64Result(0, Api::IoErrorPtr(new Network::IoSocketError(100),
+                                                     Network::IoSocketError::deleteIoError)))));
+  EXPECT_EQ(-1, bio_->method->bwrite(bio_, nullptr, 10));
+  EXPECT_EQ(ERR_GET_LIB(ERR_get_error()), ERR_LIB_SYS);
```

Comment:
Also validate the value of `getSystemErrorCode()`

Reply:
AFAICT there is no way to get the integer reason back out. I'm probably missing it? I will keep looking.

Reply:
Oh nevermind sorry I figured it out it's part of the packed error.